packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -371,23 +374,67 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
     }
   }
 
+  // TODO: Implement later. Wallets don't support this right now.
+  async signAnyTransaction(
+    transaction: GenerateTransactionInput,
+  ): Promise<AccountAuthenticator> {
+    const payloadData = transaction.data;
+    const newPayload = await TransactionBuilderTypes.generateTransactionPayload(payloadData);
+
+    if (this._wallet && !("signAnyTransaction" in this._wallet)) {
+      throw new WalletNotSupportedMethod(
+        `Sign any transaction is not supported by ${this.wallet?.name}`
+      ).message;
+    }
+    try {
+      this.doesWalletExist();
+      const response = await (this._wallet as any).signAnyTransaction(
+        newPayload
+      );
+      return response;
+    } catch (error: any) {
+      const errMsg =
+        typeof error == "object" && "message" in error ? error.message : error;
+      throw new WalletSignTransactionError(errMsg).message;
+    }
+  }
+
+  /**
+   * This function is for signing and submitting a transaction using the `@aptos-labs/ts-sdk` input types.
+   * Under the hood, it converts the input to `aptos` types and calls signAndSubmitBCSTransaction with it.
+   * 
+   * @param transactionInput the transaction input
+   * @param options max_gas_amount and gas_unit_limit
+   * @returns the response from the wallet's signAndSubmitBCSTransaction function
+   */
+  async submitTransaction(
+    transactionInput: GenerateTransactionInput,
+    options?: TransactionOptions,
+  ): Promise<any> {
+    const payloadData = transactionInput.data;
+    const newPayload = await TransactionBuilderTypes.generateTransactionPayload(payloadData);
```

Comment:
why do we need the `TransactionBuilderTypes`? can't we simply export `generateTransactionPayload` function?